source/cmake_modules/CompilationFlags.cmake

```diff
@@ -24,9 +24,9 @@ endif(NOT SKIP_TESTS)
 # Gcc Release flags
 if(CMAKE_COMPILER_IS_GNUCXX)
     set(CMAKE_GNUCXX_WARN_FLAGS "-Wall -Wextra -Woverloaded-virtual -Wundef -pedantic")
-    set(CMAKE_GNUCXX_COMMON_FLAGS "-std=c++14 -rdynamic -g -fno-omit-frame-pointer")
+    set(CMAKE_GNUCXX_COMMON_FLAGS "-std=c++14 -g -rdynamic -fno-omit-frame-pointer")
     set(CMAKE_CXX_FLAGS "${CMAKE_GNUCXX_COMMON_FLAGS} ${CMAKE_GNUCXX_WARN_FLAGS}")
-    set(CMAKE_CXX_FLAGS_PROFILE "${CMAKE_CXX_FLAG} --coverage -fprofile-arcs -ftest-coverage  -g")
+    set(CMAKE_CXX_FLAGS_PROFILE "${CMAKE_CXX_FLAG} --coverage -fprofile-arcs -ftest-coverage -O0 -g0")
```

Comment:
I am a little bit affraid about that. 
**CMAKE_CXX_FLAGS_PROFILE** is use to compile kraken but if you use -O0, optimizations  are not allowed.
I want -O0 only for sonar but not to compile kraken in an other case (like build package, classic tests on my laptop etc...)
If I am right, you need to create a CMAKE option like (-DSTRIP_SYMBOLS OR -DSKIP_TESTS to create a peculiar case for sonar) You see what I mean ?

I block the PR for the moment

Reply:
Maytbe I am false, maybe it is already an option to compile with **CMAKE_CXX_FLAGS_PROFILE**

Reply:
cmake use CMAKE_CXX_FLAG to build a project. 
But depending on the CMAKE_BUILD_TYPE=XXXX, 
cmake will add to CMAKE_CXX_FLAG flags in CMAKE_CXX_FLAG_XXXXX 
So when we build using release mode CMAKE_BUILD_TYPE=Release 
and CMAKE_CXX_FLAG contains CMAKE_CXX_FLAG_RELEASE flags BUT NOT CMAKE_CXX_FLAG_PROFILE. 
To be sure : 
we can print debug message after setting all flags : "message(${CMAKE_CXX_FLAG})", 
and we can use verbose compile mode to check all flags used by g++

Reply:
I just build in release mode (+ verbose) here are all the flags used by gcc : 
-quiet -dumpbase -mtune=generic -march=x86-64 -auxbase-strip -g -O3 -Wall -Wextra -Woverloaded-virtual -Wundef -Wpedantic -std=c++14 -version -fno-omit-frame-pointer -fuse-ld=gold -fasynchronous-unwind-tables -fstack-protector-strong -Wformat-security -fstack-clash-protection -fcf-protection

As we can see, we don't have profile flags (-O0 and -g0), but we have -g and O3 

Here we have the flags in profile mode : 
-quiet -dumpbase -mtune=generic -march=x86-64 -auxbase-strip -g -g0 -O0 -Wall -Wextra -Woverloaded-virtual -Wundef -Wpedantic -std=c++14 -version -fno-omit-frame-pointer -fuse-ld=gold -fprofile-arcs -ftest-coverage -fprofile-arcs -ftest-coverage -fprofile-update=prefer-atomic -fasynchronous-unwind-tables -fstack-protector-strong -Wformat-security -fstack-clash-protection -fcf-protection